src/lib/mongoBackend/dbFieldEncoding.cpp

```diff
@@ -41,8 +41,9 @@
 * Replace:
 *   . => =
 */
-std::string dbDotEncode(std::string s)
+std::string dbDotEncode(const std::string& _s)
```

Comment:
I tried to return "const std::string&" once and I didn't have any problem.
Perhaps it depends on how the return value is received ...

About inlining the funcion, sure.
I imagine the return/param types aren't that important anymore if he function in inline.